src/components/navBar/navBar.js

```diff
@@ -71,17 +79,20 @@ angular.module('material.components.navBar', ['material.core'])
  * `<md-nav-item>` describes a page navigation link within the `<md-nav-bar>`
  * component. It renders an md-button as the actual link.
  *
- * Exactly one of the mdNavClick, mdNavHref, mdNavSref attributes are required to be
- * specified.
+ * Exactly one of the mdNavClick, mdNavHref, mdNavSref attributes are required
+ * to be specified.
  *
  * @param {Function=} mdNavClick Function which will be called when the
- * link is clicked to change the page. Renders as an `ng-click`.
+ *     link is clicked to change the page. Renders as an `ng-click`.
  * @param {string=} mdNavHref url to transition to when this link is clicked.
- * Renders as an `ng-href`.
+ *     Renders as an `ng-href`.
  * @param {string=} mdNavSref Ui-router state to transition to when this link is
```

Comment:
I am not responsible for the initial implementation here, but I believe that the `md-nav-bar` is meant to act as you primary navigation system and needs to support `$router` and `ui-router` options. The HTML that you write is translated into `md-button` templates that have the `$router` and/or `ui-router` attributes added to them.